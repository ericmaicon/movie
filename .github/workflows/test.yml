name: Lint & Test

on: push

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
        with:
          node-version: '14'
      - name: Cache dependencies
        uses: actions/cache@v2.1.4
        id: cache-dependencies
        with:
          path: |
            node_modules
            */*/node_modules
          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
      - name: Install Dependencies
        if: steps.cache-dependencies.outputs.cache-hit != 'true'
        run: yarn install && yarn bootstrap
      - name: Link
        run: yarn lerna link
      - name: Build deps
        run: yarn build
      - name: Lint
        run: yarn lint
      - name: Run Tests
        run: yarn test
      - uses: codecov/codecov-action@v1
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          directory: packages/**/coverage
          fail_ci_if_error: true
