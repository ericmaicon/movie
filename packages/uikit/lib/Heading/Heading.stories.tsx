import React from 'react';

import { Story, Meta } from '@storybook/react';

import Heading, { HeadingProps } from './Heading';

export default {
  title: 'Typography/Heading',
  component: Heading,
} as Meta;

const Template: Story<HeadingProps> = (args) => <Heading {...args} />;

export const Heading1 = Template.bind({});
Heading1.args = {
  children: 'Heading 1',
};

export const Heading2 = Template.bind({});
Heading2.args = {
  children: 'Heading 2',
  size: 2,
};

export const Heading3 = Template.bind({});
Heading3.args = {
  children: 'Heading 3',
  size: 3,
};

export const Heading4 = Template.bind({});
Heading4.args = {
  children: 'Heading 4',
  size: 4,
};

export const Heading5 = Template.bind({});
Heading5.args = {
  children: 'Heading 5',
  size: 5,
};
