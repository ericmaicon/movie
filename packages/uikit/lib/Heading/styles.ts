import styled from 'styled-components';

export type StyledHeadingProps = {
  /**
   * How large should the heading be?
   */
  size: 1 | 2 | 3 | 4 | 5;
};

export const Heading1 = styled.h1<StyledHeadingProps>`
  margin: 0px;
  font-size: 2.5rem;
  ${({ theme }) => theme.typography.heading};

  //tablet
  @media (min-width: 768px) and (max-width: 1024px) {
    font-size: 3rem;
  }

  //desktop
  @media (min-width: 1025px) {
    font-size: 4rem;
  }
`;

export const Heading2 = styled.h2<StyledHeadingProps>`
  margin: 0px;
  font-size: 2rem;
  ${({ theme }) => theme.typography.heading};

  //tablet
  @media (min-width: 768px) and (max-width: 1024px) {
    font-size: 2.5rem;
  }

  //desktop
  @media (min-width: 1025px) {
    font-size: 3rem;
  }
`;

export const Heading3 = styled.h3<StyledHeadingProps>`
  margin: 0px;
  font-size: 1.5rem;
  ${({ theme }) => theme.typography.heading};

  //tablet
  @media (min-width: 768px) and (max-width: 1024px) {
    font-size: 2rem;
  }

  //desktop
  @media (min-width: 1025px) {
    font-size: 2.5rem;
  }
`;

export const Heading4 = styled.h4<StyledHeadingProps>`
  margin: 0px;
  font-size: 1rem;
  ${({ theme }) => theme.typography.heading};

  //tablet
  @media (min-width: 768px) and (max-width: 1024px) {
    font-size: 1.5rem;
  }

  //desktop
  @media (min-width: 1025px) {
    font-size: 2rem;
  }
`;

export const Heading5 = styled.h4<StyledHeadingProps>`
  margin: 0px;
  font-size: 0.5rem;
  ${({ theme }) => theme.typography.heading};

  //tablet
  @media (min-width: 768px) and (max-width: 1024px) {
    font-size: 1rem;
  }

  //desktop
  @media (min-width: 1025px) {
    font-size: 1.5rem;
  }
`;
