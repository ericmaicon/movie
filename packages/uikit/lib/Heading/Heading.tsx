import React, { BaseHTMLAttributes, ReactNode } from 'react';

import * as Styled from './styles';

export interface HeadingProps
  extends BaseHTMLAttributes<HTMLHeadingElement>,
    Partial<Styled.StyledHeadingProps> {
  /**
   * Heading content.
   */
  children: ReactNode;
}

const Heading = ({ children, size = 1, ...rest }: HeadingProps) => {
  switch (size) {
    case 1:
    default:
      return (
        <Styled.Heading1 size={size} {...rest}>
          {children}
        </Styled.Heading1>
      );
    case 2:
      return (
        <Styled.Heading2 size={size} {...rest}>
          {children}
        </Styled.Heading2>
      );
    case 3:
      return (
        <Styled.Heading3 size={size} {...rest}>
          {children}
        </Styled.Heading3>
      );
    case 4:
      return (
        <Styled.Heading4 size={size} {...rest}>
          {children}
        </Styled.Heading4>
      );
    case 5:
      return (
        <Styled.Heading5 size={size} {...rest}>
          {children}
        </Styled.Heading5>
      );
  }
};

export default Heading;
