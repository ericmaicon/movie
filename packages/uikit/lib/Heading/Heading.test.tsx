import React from 'react';

import { render } from '~/test/index';

import Heading from './Heading';

describe('Heading', () => {
  it('should render correctly', () => {
    const { getByRole } = render(<Heading>Heading</Heading>);

    const headingElement = getByRole('heading');

    expect(headingElement).toBeInTheDocument();
    expect(headingElement).toHaveTextContent(/^Heading$/);
    expect(headingElement).toMatchSnapshot();
  });

  it('should render a heading 1', () => {
    const { getByRole } = render(<Heading size={1}>Heading</Heading>);

    const headingElement = getByRole('heading');
    expect(headingElement).toBeInTheDocument();
    expect(headingElement).toMatchSnapshot();
  });

  it('should render a heading 2', () => {
    const { getByRole } = render(<Heading size={2}>Heading</Heading>);

    const headingElement = getByRole('heading');
    expect(headingElement).toBeInTheDocument();
    expect(headingElement).toMatchSnapshot();
  });

  it('should render a heading 3', () => {
    const { getByRole } = render(<Heading size={3}>Heading</Heading>);

    const headingElement = getByRole('heading');
    expect(headingElement).toBeInTheDocument();
    expect(headingElement).toMatchSnapshot();
  });

  it('should render a heading 4', () => {
    const { getByRole } = render(<Heading size={4}>Heading</Heading>);

    const headingElement = getByRole('heading');
    expect(headingElement).toBeInTheDocument();
    expect(headingElement).toMatchSnapshot();
  });

  it('should render a heading 5', () => {
    const { getByRole } = render(<Heading size={5}>Heading</Heading>);

    const headingElement = getByRole('heading');
    expect(headingElement).toBeInTheDocument();
    expect(headingElement).toMatchSnapshot();
  });
});
