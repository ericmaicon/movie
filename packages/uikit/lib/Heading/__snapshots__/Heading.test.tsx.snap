// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render a heading 1 1`] = `
.c0 {
  margin: 0px;
  font-size: 2.5rem;
  font-family: Roboto;
  font-weight: 700;
  font-size: 4rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
}

@media (min-width:768px) and (max-width:1024px) {
  .c0 {
    font-size: 3rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    font-size: 4rem;
  }
}

<h1
  class="c0"
  size="1"
>
  Heading
</h1>
`;

exports[`Heading should render a heading 2 1`] = `
.c0 {
  margin: 0px;
  font-size: 2rem;
  font-family: Roboto;
  font-weight: 700;
  font-size: 4rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
}

@media (min-width:768px) and (max-width:1024px) {
  .c0 {
    font-size: 2.5rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    font-size: 3rem;
  }
}

<h2
  class="c0"
  size="2"
>
  Heading
</h2>
`;

exports[`Heading should render a heading 3 1`] = `
.c0 {
  margin: 0px;
  font-size: 1.5rem;
  font-family: Roboto;
  font-weight: 700;
  font-size: 4rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
}

@media (min-width:768px) and (max-width:1024px) {
  .c0 {
    font-size: 2rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    font-size: 2.5rem;
  }
}

<h3
  class="c0"
  size="3"
>
  Heading
</h3>
`;

exports[`Heading should render a heading 4 1`] = `
.c0 {
  margin: 0px;
  font-size: 1rem;
  font-family: Roboto;
  font-weight: 700;
  font-size: 4rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
}

@media (min-width:768px) and (max-width:1024px) {
  .c0 {
    font-size: 1.5rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    font-size: 2rem;
  }
}

<h4
  class="c0"
  size="4"
>
  Heading
</h4>
`;

exports[`Heading should render a heading 5 1`] = `
.c0 {
  margin: 0px;
  font-size: 0.5rem;
  font-family: Roboto;
  font-weight: 700;
  font-size: 4rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
}

@media (min-width:768px) and (max-width:1024px) {
  .c0 {
    font-size: 1rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    font-size: 1.5rem;
  }
}

<h4
  class="c0"
  size="5"
>
  Heading
</h4>
`;

exports[`Heading should render correctly 1`] = `
.c0 {
  margin: 0px;
  font-size: 2.5rem;
  font-family: Roboto;
  font-weight: 700;
  font-size: 4rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
}

@media (min-width:768px) and (max-width:1024px) {
  .c0 {
    font-size: 3rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    font-size: 4rem;
  }
}

<h1
  class="c0"
  size="1"
>
  Heading
</h1>
`;
